pkg/logs/launchers/integration/launcher.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package integration creates a launcher to track logs from integrations
+package integration
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	ddLog "github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	"github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+	pkgConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/launchers"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	"github.com/DataDog/datadog-agent/pkg/logs/tailers"
+)
+
+// Launcher checks for launcher integrations, creates files for integrations to
+// write logs to, then creates file sources for the file launcher to tail
+type Launcher struct {
+	sources              *sources.LogSources
+	piplineProvider      pipeline.Provider
+	registry             auditor.Registry
+	addedSources         chan *sources.LogSource
+	removedSources       chan *sources.LogSource
```

Comment:
Removed the unused `removedSources`, `pipelineProvider`, and `registry` parameters. The `pipeline.Provider` and `auditor.Registry` types are needed in the `Launcher.Start()` function because they implement the `Launcher` interface, so I just made the unnamed parameters instead.